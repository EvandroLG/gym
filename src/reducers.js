import { combineReducers } from 'redux';
import { SET_VIDEO_URL } from './actions';

const videoUrl = (state='', action) => {
  if (action.type === SET_VIDEO_URL) {
    return action.payload;
  }

  return state;
};

export default combineReducers({
  videoUrl
});
