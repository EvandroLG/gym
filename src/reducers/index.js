import { combineReducers } from 'redux';
import videoUrl from './video_url';
import newTraining from './new_training';

export default combineReducers({
  videoUrl,
  newTraining
});
