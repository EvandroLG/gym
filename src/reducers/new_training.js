const defaultState = {
  title: '',
  exerciseList: []
};

const newTraining = (state = defaultState, action) => {
  if (action.type === 'SET_TITLE_TRAINING') {
    return { ...state, title: action.title };
  }

  if (action.type === 'ADD_EXERCISE_TRAINING') {
    const defaultExercise = {
      name: '',
      repetition: '',
      set: '',
      weight: '',
      youtube: ''
    };

    const id = state.exerciseList.length;
    const newExercise = { ...defaultExercise, id };
    const exerciseList = state.exerciseList.concat(newExercise);

    return { ...state, exerciseList };
  }

  if (action.type === 'REMOVE_EXERCISE_TRAINING') {
    const exerciseList = state.exerciseList.filter((exercise) => {
      return exercise.id !== action.id;
    });

    return { ...state, exerciseList };
  }

  if (action.type === 'SET_FIELD_TRAINING') {
    const exerciseList = state.exerciseList.map((exercise) => {
      if (exercise.id === action.id) {
        exercise[action.fieldName] = action.value;
      }

      return exercise;
    });

    return { ...state, exerciseList };
  }

  return state;
};

export default newTraining;
