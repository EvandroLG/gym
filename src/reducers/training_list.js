const defaultState = [];

const trainingList = (state = defaultState, action) => {
  if (action.type === 'LOAD_TRAINING_LIST') {
    return action.trainingList;
  }

  if (action.type === 'ADD_NEW_EXERCISE') {
    const defaultExercise = {
      name: '',
      repetition: '',
      set: '',
      weight: '',
      youtube: ''
    };

    return state.map((training) => {
      if (training.id === action.id) {
        const id = training.exerciseList.length;
        const newExercise = { ...defaultExercise, id };
        training.exerciseList.push(newExercise);

        return training;
      }

      return training;
    });
  }

  if (action.type === 'REMOVE_EXERCISE') {
    return state.map((training) => {
      if (training.id === action.idTraining) {
        const exerciseList = training.exerciseList.filter((exercise) => {
          if (exercise.id !== action.idExercise) {
            return exercise;
          }
        });

        return { ...training, exerciseList };
      }

      return training;
    });
  }

  if (action.type === 'CHANGE_TITLE_TRAINING') {
    return state.map((training) => {
      if (training.id === action.id) {
        training.title = action.title;
      }

      return training;
    });
  }

  if (action.type === 'CHANGE_EXERCISE_TRAINING') {
    return state.map((training) => {
      if (training.id === action.idTraining) {
        const exerciseList = training.exerciseList.map((exercise) => {
          if (exercise.id === action.idExercise) {
            exercise[action.property] = action.value;
          }

          return exercise;
        });

        training.exerciseList = exerciseList;
      }

      return training;
    });
  }

  return state;
};

export default trainingList;
