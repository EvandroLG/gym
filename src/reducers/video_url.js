const videoUrl = (state = '', action) => {
  if (action.type === 'SET_VIDEO_URL') {
    return action.url;
  }

  return state;
};

export default videoUrl;
