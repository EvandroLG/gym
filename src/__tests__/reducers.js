import reducers from '../reducers';
import fixtures from './fixtures';

describe('reducers', () => {

  const initialFakeState = {
    videoUrl: '',
    trainingList: [],
    newTraining: fixtures[0]
  };

  function verifyReducers(action, state) {
    expect(reducers(state, action)).toMatchSnapshot();
  }

  describe('video url', () => {
    it('should handle SET_VIDEO_URL', () => {
      verifyReducers({
        type: 'SET_VIDEO_URL',
        url: 'https://www.youtube.com/watch?v=04p4MWfhpAI'
      });
    });
  });

  describe('new training', () => {
    it('should handle SET_TITLE_NEW_TRAINING', () => {
      const { title } = fixtures[0];

      verifyReducers({
        type: 'SET_TITLE_NEW_TRAINING',
        title
      });
    });

    it('should handle ADD_EXERCISE_NEW_TRAINING', () => {
      verifyReducers({
        type: 'ADD_EXERCISE_NEW_TRAINING',
      });
    });

    it('should handle REMOVE_EXERCISE_NEW_TRAINING', () => {
      verifyReducers({
        type: 'REMOVE_EXERCISE_NEW_TRAINING',
        id: 0
      }, initialFakeState);
    });

    it('should handle SET_FIELD_NEW_TRAINING', () => {
      verifyReducers({
        type: 'SET_FIELD_NEW_TRAINING',
        id: 0,
        fieldName: 'name',
        value: 'Pull-Down'
      }, initialFakeState);
    });

    it('should handle EMPTY_NEW_TRAINING', () => {
      verifyReducers({
        type: 'EMPTY_NEW_TRAINING',
      }, initialFakeState);
    });
  });

  describe('training list', () => {
    it('should handle LOAD_TRAINING_LIST', () => {
      verifyReducers({
        type: 'LOAD_TRAINING_LIST',
        trainingList: []
      });
    });

    it('should handle REMOVE_TRAINING', () => {
      verifyReducers({
        type: 'REMOVE_TRAINING',
        id: 0
      }, initialFakeState);
    });

    it('should handle ADD_NEW_EXERCISE', () => {
      verifyReducers({
        type: 'ADD_NEW_EXERCISE',
        id: 1
      }, initialFakeState);
    });

    it('should handle REMOVE_EXERCISE', () => {
      verifyReducers({
        type: 'REMOVE_EXERCISE',
        idTraining: 1,
        idExercise: 0
      }, initialFakeState);
    });

    it('should handle CHANGE_TITLE_TRAINING', () => {
      verifyReducers({
        type: 'CHANGE_TITLE_TRAINING',
        id: 1,
        title: 'Arms'
      }, initialFakeState);
    });

    it('should handle CHANGE_EXERCISE_TRAINING', () => {
      verifyReducers({
        type: 'CHANGE_EXERCISE_TRAINING',
        idTraining: 1,
        idExercise: 0,
        value: 'Pull-Down'
      }, initialFakeState);
    });
  });
});
