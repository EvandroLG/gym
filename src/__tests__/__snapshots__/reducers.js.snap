// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers should return an object with the same value passed as a parameter 1`] = `
Object {
  "newTraining": Object {
    "exerciseList": Array [],
    "title": "",
  },
  "trainingList": Array [],
  "videoUrl": "https://www.youtube.com/watch?v=04p4MWfhpAI",
}
`;
